storage/tick.go

```diff
@@ -125,36 +143,41 @@ func (mgr *tickManager) Tick(softDeadline time.Duration, forceType forceType) er
 
 	// Begin ticking
 	start := mgr.nowFn()
-	sizes := make([]int64, 0, len(namespaces))
-	totalSize := int64(0)
 	for _, n := range namespaces {
-		size := n.NumSeries()
-		sizes = append(sizes, size)
-		totalSize += size
+		n.Tick(mgr.c)
 	}
-	for i, n := range namespaces {
-		deadline := float64(softDeadline) * (float64(sizes[i]) / float64(totalSize))
-		n.Tick(mgr.c, time.Duration(deadline))
+
+	// NB(r): Always sleep for some constant period since ticking
+	// is variable with num series. With a really small amount of series
+	// the per shard amount of constant sleeping is only:
+	// = num shards * sleep per series (100 microseconds default)
+	// This number can be quite small if configuring to have small amount of
+	// shards (say 10 shards) with 32 series per shard (total of 320 series):
+	// = 10 num shards * ~32 series per shard * 100 microseconds default sleep per series
+	// = 10*32*0.1 milliseconds
+	// = 32ms
+	// Because of this we always sleep at least some fixed constant amount.
+	took := mgr.nowFn().Sub(start)
+	mgr.metrics.tickWorkDuration.Record(took)
+
+	mgr.runtimeOpts.RLock()
```

Comment:
nit: could you make this get a method on the runtimeOpts struct

Reply:
Sure thing.